src/ServicePulse.Host/vue/src/composables/serviceLicense.js

```diff
@@ -40,16 +40,19 @@ export const licenseStatus = reactive({
 });
 
 export function useLicense() {
-  watch(license, async (newValue, oldValue) => {
-    const checkForWarnings = oldValue !== null ? newValue && newValue.license_status != oldValue.license_status : newValue !== null;
-    if (checkForWarnings) {
-      displayWarningMessage(newValue.license_status);
+  watch(
+    () => license.license_status,
+    async (newValue, oldValue) => {
+      const checkForWarnings = oldValue !== null ? newValue && newValue != oldValue : newValue !== null;
+      if (checkForWarnings) {
+        displayWarningMessage(newValue);
+      }
     }
-  });
+  );
 
   return getLicense(license)
     .then((lic) => {
-      license = lic;
+      Object.assign(license, lic);
```

Comment:
Is there any specific reason, you had use the Object.assign?

Reply:
Yes, because the assignment operator `=` creates a new reference to the original object, which is not reactive. If you want to create a reactive copy of an object, we should be using `Object.assign` or the spread operator `...`

https://stackoverflow.com/questions/65732144/vue-js-3-replace-update-reactive-object-without-losing-reactivity